src/apps/polymer/vIBCEscrow.sol

```diff
@@ -0,0 +1,181 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+import { APolymerEscrow } from "./APolymerEscrow.sol";
+import "../../MessagePayload.sol";
+
+import  "vibc-core-smart-contracts/interfaces/IbcDispatcher.sol";
+import { AckPacket } from "vibc-core-smart-contracts/libs/Ibc.sol";
+import { IbcReceiverBase, IbcReceiver } from "vibc-core-smart-contracts/interfaces/IbcReceiver.sol";
+
+/** 
+ * @notice Polymer implementation of the Generalised Incentives based on vIBC.
+ * @dev An implementation quirk of Polymer is that channels map 1:1 to BOTH chains
+ * and contracts. As a result, if we trust a channel, we also imply that we trust
+ * the contract on the other end of that channel. This is unlike "traditional" chain
+ * mappings where there may be may addresses on the other end.
+ * As a result, we are allowed to just append our address to the package and then trust that.
+ * Because if someone trust the channel (which is a requirement) then they must also
+ * trust the account AND the set value.
+ */
+contract IncentivizedPolymerEscrow is APolymerEscrow, IbcReceiverBase, IbcReceiver {
+    error ChannelNotFound();
+    error UnsupportedVersion();
+
+    uint constant POLYMER_SENDER_IDENTIFIER_START = 0;
+    uint constant POLYMER_SENDER_IDENTIFIER_END = 32;
+    uint constant POLYMER_PACKAGE_PAYLOAD_START = 32;
+
+    bytes32[] public connectedChannels;
+    string constant VERSION = 'vIBC Escrow 1.0';
```

Comment:
Channel version needs to exactly `"1.0"` for now due to constraints on our side. But we will make it more flexible in the future